src/wp-includes/formatting.php

```diff
@@ -3914,10 +3914,22 @@ function wp_trim_excerpt( $text = '', $post = null ) {
 		$text = strip_shortcodes( $text );
 		$text = excerpt_remove_blocks( $text );
 
+		/*
+		 * Temporarily unhook wp_filter_content_tags() since any tags
+		 * within the excerpt are stripped out. Modifying the tags here
+		 * is wasteful and can lead to bugs in the image counting logic.
+		 */
+		$filter_removed = remove_filter( 'the_content', 'wp_filter_content_tags' );
```

Comment:
@felixarntz Is it possible that we could ever change the priority that `wp_filter_content_tags()` is hooked to `'the_content'`?

Just wondering if we should be storing the result of `has_filter()` here, and if `is_int( $has_filter )`, then remove the filter. Then when restoring, we can pass the priority back into the `add_filter()` call.

Reply:
@costdev That seems unnecessary to me as WP core is in control about the filter priority so if we were to change it in the future (which is unlikely as that would be a BC break), we could update it in both places.

Yes, it makes the code a bit more flexible, but it also adds another check which IMO isn't really necessary.

Reply:
+1 to @felixarntz comment. It is a worry, but it is an edge case. Something to flag in the dev note.